### PR TITLE
Add --no-creds flag to skopeo inspect

### LINKED ISSUE #421: RFE: Allow force anonymous inspect
By default skopeo tries to use ~/.docker/config.json to authenticate to an external registry unless username and password are provided using --creds flag

It would be handy to allow to force no token to use skopeo as a debug tool

### PR DESCRIPTION
**Pull request by QiWang19**
Follow PR #433 
Close #421 
Currently skopeo inspect allows to:

- Use the default credentials in $HOME/.docker.config
- Explicitly define credentials via de --creds flag
This implements a --no-creds flag which will query docker registries anonymously.

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

File: cmd/skopeo/copy.go [hunk 1]
```diff
@@ -24,7 +24,8 @@ type copyOptions struct {
 	signByFingerprint string          // Sign the image using a GPG key with the specified fingerprint
 	format            optionalString  // Force conversion of the image to a specified format
 	quiet             bool            // Suppress output information when copying images
-
+	srcNoCreds        bool
+	destNoCreds       bool
```
Comment (mtrmac) [inline cmd/skopeo/copy.go]: This seems unused, along with all the other variables added outside of `utils.go`.

File: cmd/skopeo/inspect.go [hunk 2]
```diff
@@ -58,7 +59,7 @@ func inspectCmd(global *globalOptions) cli.Command {
 		Flags: append(append([]cli.Flag{
 			cli.BoolFlag{
 				Name:        "raw",
-				Usage:       "output raw manifest",
+				Usage:       "Output raw manifest",
```
Comment (mtrmac) [inline cmd/skopeo/inspect.go]: If not a separate PR, make this at least a separate commit, please.

**Comment by TomSweeneyRedHat** [comment]
OK, I might not be remembering correctly, but I thought that `skopeo inspect` worked against a registry that didn't require creds already?  If so, why do we need the `--no-creds` flag?

**Comment by mtrmac** [comment]
> why do we need the `--no-creds` flag?

See https://github.com/containers/skopeo/pull/433#issuecomment-337062649 .

**Comment by TomSweeneyRedHat** [comment]
Thanks @mtrmac.  Prior to this patch, could they just `docker/podman logout` and then pull successfully?

File: cmd/skopeo/inspect.go (hunk 2 above)
Comment (QiWang19) [reply cmd/skopeo/inspect.go]: maybe in another PR.

**Comment by QiWang19** [comment]
to document `c/image/types.DockerAuthConfig` can be used with empty fields, just change the comment in `c/image/types/types.go` ?

**Comment by mtrmac** [comment]
> to document `c/image/types.DockerAuthConfig` can be used with empty fields, just change the comment in `c/image/types/types.go` ?

Yes; we just need such a PR to exist in c/image, this PR does not need to block on it nor vendor an updated version of c/image just for the comment.

File: vendor/github.com/containers/image/docs/atomic-signature-embedded-json.json [hunk 3]
```diff
@@ -1,66 +0,0 @@
-{
```
Comment (mtrmac) [inline vendor/github.com/containers/image/docs/atomic-signature-embedded-json.json]: Why does this PR remove the various files in `vendor`?

(Guessing, run `make vendor` instead of `vndr` directly.)

**Comment by QiWang19** [comment]
can this get in?

**Comment by mtrmac** [comment]
LGTM.  Thanks!
